Essentials/src/com/earth2me/essentials/commands/Commandafk.java

```diff
@@ -59,25 +59,33 @@ private void toggleAfk(User sender, User user, String message) throws Exception
         }
         user.setDisplayNick();
         String msg = "";
+        String selfmsg = "";
         if (!user.toggleAfk()) {
-            //user.sendMessage(_("markedAsNotAway"));
             if (!user.isHidden()) {
                 msg = tl("userIsNotAway", user.getDisplayName());
+                selfmsg = tl("userIsNotAwaySelf");
             }
             user.updateActivity(false);
         } else {
-            //user.sendMessage(_("markedAsAway"));
             if (!user.isHidden()) {
                 if (message != null) {
                     msg = tl("userIsAwayWithMessage", user.getDisplayName(), message);
+                    selfmsg = tl("userIsAwaySelfWithMessage", message);
                 } else {
                     msg = tl("userIsAway", user.getDisplayName());
+                    selfmsg = tl("userIsAwaySelf");
                 }
             }
             user.setAfkMessage(message);
         }
         if (!msg.isEmpty()) {
```

Comment:
So my interpretation of this is dual meaning - if the player previously went in and made userIs.*Away.* messages empty, they would not be displayed. I'm not sure if this is intended behavior or not, but this should probably be preserved for selfmsg as well.

Bonus points in figuring out if broadcastAfkMessage is true yet selfmsg is empty what the right thing to do is (you could display the broadcast message to the user, but maybe the server owner wanted no message sent to the user?)

Reply:
I think that the best thing to do here is to let the server user have full control over whether how the message will be sent. If they don't want any message sent, then we should let them. Otherwise, they could just copy the message key from `userIsAway` to `userIsAwaySelf` if that's what they'd prefer.